### PR TITLE
Disable HTTPS & shift functions to block page

### PR DESCRIPTION
**Pull request by WaLLy3K**
**By submitting this pull request, I confirm the following:**

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [x] I have checked that [another pull request](https://github.com/pi-hole/pi-hole/pulls) for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

**How familiar are you with the codebase?:**

5

---

The changes here primarily shift the work of header functions, JS redirects, etc, to the block page as per https://github.com/pi-hole/pi-hole/pull/1416.

Also, by commenting out `include-conf-enabled.pl` and replacing it with `include_shell "find /etc/lighttpd/conf-enabled -name '*.conf' -a ! -name 'letsencrypt.conf' -printf 'include \"%p\"\n' 2>/dev/null"`, the following can be added to `external.conf` to enable HTTPS, without affecting web browsing load speeds:
```
# Only enable SSL for primary FQDN
$HTTP["host"] == "rpi.ddns.com" {
  include_shell "cat conf-enabled/letsencrypt.conf 2>/dev/null"
}
```

A blocked HTTPS query will be instantly dropped, but may provide an error message similar to: `curl: (35) error:14077458:SSL routines:SSL23_GET_SERVER_HELLO:tlsv1 unrecognized name`. Autofilling `$HTTP["host"] == ""` within `external.conf` with the proposed `FQDN` key from `setupVars.conf` is not a part of this PR and can be developed separately.

### CODE REVIEW

**Comment by PromoFaux** [comment]
@WaLLy3K what steps do I need to take to test that this is working? So far it doesn't appear to be affecting any normal operations, but what should I be looking for specifically?

**Comment by WaLLy3K** [comment]
As long as you get this error message when you check the headers of a blocked HTTPS site, then it's working as expected:

```
curl -I https://doubleclick.net
curl: (35) error:14077458:SSL routines:SSL23_GET_SERVER_HELLO:tlsv1 unrecognized name
```

**Comment by PromoFaux** [comment]
I get this (and doubleclick is def blocked)

```
fap@fappotron:/etc/.pihole$ curl -I https://doubleclick.net
curl: (7) Failed to connect to doubleclick.net port 443: Connection refused
fap@fappotron:/etc/.pihole$ nslookup doubleclick.net
Server:         127.0.0.1
Address:        127.0.0.1#53

Name:   doubleclick.net
Address: 192.168.0.2

```

**Comment by PromoFaux** [comment]
I would approve this, but I'm not set up with HTTPS on my pi-hole installation (and quite frankly, I don't have a clue how to go about getting my setup into a state to be able to fully test this) Logic seems sound, however.

**Comment by spacedingo** [comment]
I have been tested this type of "breaking" the https request and the results are great with respect to page-load times. Those HTTPS error codes are the exact ones I got after I configured nginx and it was flying. I may have screenshots of browser dev-tools page load times before and after. If it can be done with lighttpd, then all the better.

**Comment by spacedingo** [comment]
The following curl fetch times were taken with regular pi-hole and the a pi-hole with nginx configure to give the same https error (35) as above.

Fail to connect:
v4 http 0.031000
v6 http 0.000000
v4 https 0.204000
v6 https 0.203000

Protocol error:
v4 http 0.000000
v6 http 0.031000
v4 https 0.094000
v6 https 0.110000

The https response times are halved. It does cut a 5-6 second page load time to 2-3 seconds from what I have seen, considering that there are so few http content. Sorry I'm using nginx - i didn't know how to configure lighttpd for the same result.

Regular pi-hole (fail to connect)
```
G:\>curl -4 -I -w %{time_total} http://scorecardresearch.com/somethingsomething.jpg
HTTP/1.1 200 OK
Content-type: text/html; charset=UTF-8
Date: Wed, 16 Aug 2017 03:45:07 GMT
Server: lighttpd/1.4.45

0.031000

G:\>curl -4 -I -w %{time_total} https://scorecardresearch.com/somethingsomething.jpg
0.204000curl: (7) Failed to connect to scorecardresearch.com port 443: Connection refused

G:\>curl -6 -I -w %{time_total} http://scorecardresearch.com/somethingsomething.jpg
HTTP/1.1 200 OK
Content-type: text/html; charset=UTF-8
Date: Wed, 16 Aug 2017 03:45:48 GMT
Server: lighttpd/1.4.45

0.000000

G:\>curl -6 -I -w %{time_total} https://scorecardresearch.com/somethingsomething.jpg
0.203000curl: (7) Failed to connect to scorecardresearch.com port 443: Connection refused

G:\>curl -6 -I -w %{time_total} https://scorecardresearch.com/somethingsomething.jpg
0.218000curl: (7) Failed to connect to scorecardresearch.com port 443: Connection refused
```
Pihole/nginx (error 35)
```
G:\>curl -4 -I -w %{time_total} http://scorecardresearch.com/somethingsomething.jpg
HTTP/1.1 200 OK
Server: nginx/1.10.3
Date: Wed, 16 Aug 2017 03:39:12 GMT
Content-Type: image/gif
Content-Length: 43
Last-Modified: Mon, 28 Sep 1970 06:00:00 GMT
Connection: close
Expires: Wed, 16 Aug 2017 03:39:11 GMT
Cache-Control: no-cache
Strict-Transport-Security: includeSubdomains;
X-Frame-Options: DENY
X-Content-Type-Options: nosniff

0.000000
G:\>curl -4 -I -w %{time_total} https://scorecardresearch.com/somethingsomething.jpg
0.094000curl: (35) Unknown SSL protocol error in connection to scorecardresearch.com:443

G:\>curl -6 -I -w %{time_total} http://scorecardresearch.com/somethingsomething.jpg
HTTP/1.1 200 OK
Server: nginx/1.10.3
Date: Wed, 16 Aug 2017 03:39:38 GMT
Content-Type: image/gif
Content-Length: 43
Last-Modified: Mon, 28 Sep 1970 06:00:00 GMT
Connection: close
Expires: Wed, 16 Aug 2017 03:39:37 GMT
Cache-Control: no-cache
Strict-Transport-Security: includeSubdomains;
X-Frame-Options: DENY
X-Content-Type-Options: nosniff

0.031000
G:\>curl -6 -I -w %{time_total} https://scorecardresearch.com/somethingsomething.jpg
0.110000curl: (35) Unknown SSL protocol error in connection to scorecardresearch.com:443
```

File: advanced/lighttpd.conf.debian [hunk 1]
```diff
@@ -59,21 +60,9 @@ $HTTP["url"] =~ "^/admin/" {
...
     )
-}
-
-# Rewite js requests, must be out of $HTTP block due to bug #2526
-url.rewrite = ( "^(?!/admin/).*\.js$"  => "pihole/index.js"   )
-
-# If the URL does not start with /admin, then it is a query for an ad domain
-$HTTP["url"] =~ "^(?!/admin)/.*" {
-    # Create a response header for debugging using curl -I
-    setenv.add-response-header = ( "X-Pi-hole" => "A black hole for Internet advertisements." )
```
Comment (Mcat12) [inline advanced/lighttpd.conf.debian]: I think we want to keep the X-Pi-hole header for non-admin queries (it's being checked for in the dev debug script).

Comment (WaLLy3K) [inline advanced/lighttpd.conf.debian]: IIRC it's replaced by [this codeblock](https://github.com/WaLLy3K/pi-hole/blob/78143c2ff41c2088b7db9efbeb882a62de944ef3/advanced/index.php#L46).

**Comment by PromoFaux** [comment]
Is this one good to go? I still don't really get it, so am unable to test... so we may have to go on trust on this one... AFAIK, this is the only one blocking BlockPage 2.0?

**Comment by WaLLy3K** [comment]
There's not much to it really, but this requires features that the block page needs to add first.

**Comment by PromoFaux** [comment]
Approved

**Pull request merged**